agent/src/beerocks/fronthaul_manager/ap_manager/ap_manager_thread.h

```diff
@@ -35,6 +35,8 @@ class ap_manager_thread : public beerocks::socket_thread {
         bool iface_filter_low;
     };
 
+    enum class eApManagerState { INIT, WAIT_FOR_CONFIGURATION, ATTACHING, ATTACHED, OPERATIONAL };
```

Comment:
Please add a doxygen comment explaining each state.

Reply:
I think that the state name explains itself, and that trying to explain it will look ridiculous since I would have explain it with the state name. What's done in each state can be understood by looking at what is being done on the fsm function.

All that said, I think that documenting an enum/struct is weird and we should not do it.
We need to document what is not obvious like functions and logic. Documenting any other thing is redundant.

Reply:
Maybe in this case the cases are self-explanatory, but in other places this may not be the case.
Documentation is not meant for the developer writing it, but to others that may read the code (or even the generated doxygen file, and in this case they don't even have the source code to look at).